internal/pkg/githubapi/github.go

```diff
@@ -230,6 +226,50 @@ func HandlePREvent(eventPayload *github.PullRequestEvent, ghPrClientDetails GhPr
 	}
 }
 
+func generateArgoCdDiffComments(diffCommentData DiffCommentData, githubCommentMaxSize int) (comments []string, err error) {
```

Comment:
I personally think this function is over-commented, and some of those comments can go to the function comment.